manifests/gui.pp

```diff
@@ -2,65 +2,98 @@
+  $admin_users          = $::icinga::admin_users,
+  $admin_group          = $::icinga::admin_group,
+  $ro_users             = $::icinga::ro_users,
+  $ro_group             = $::icinga::ro_group,
+  $manage_ssl           = $::icinga::manage_ssl,
+  $gui_type             = $::icinga::gui_type,
+  $nagios_plugins       = $::icinga::nagios_plugins,
+  $ido_db_server        = $::icinga::ido_db_server,
+  $ido_db_user          = $::icinga::ido_db_user,
+  $ido_db_pass          = $::icinga::ido_db_pass,
+  $ido_db_host          = $::icinga::ido_db_host,
+  $ido_db_port          = $::icinga::ido_db_port,
+  $ido_db_name          = $::icinga::ido_db_name,
+  $web_auth_type        = $::icinga::web_auth_type,
+  $web_auth_name        = $::icinga::web_auth_name,
+  $web_auth_user_file   = $::icinga::web_auth_user_file,
+  $web_auth_group_file  = $::icinga::web_auth_group_file,
+  $web_auth_users       = $::icinga::web_auth_users,
+  $web_db_server        = $::icinga::web_db_server,
+  $web_db_user          = $::icinga::web_db_user,
+  $web_db_pass          = $::icinga::web_db_pass,
+  $web_db_host          = $::icinga::web_db_host,
+  $web_db_port          = $::icinga::web_db_port,
+  $web_db_name          = $::icinga::web_db_name,
+  $ldap_security        = $::icinga::ldap_security,
+  $ldap_server          = $::icinga::ldap_server,
+  $ldap_firstname       = $::icinga::ldap_firstname,
+  $ldap_lastname        = $::icinga::ldap_lastname,
+  $ldap_email           = $::icinga::ldap_email,
+  $ldap_basedn          = $::icinga::ldap_basedn,
+  $ldap_groupdn         = $::icinga::ldap_groupdn,
+  $ldap_binddn          = $::icinga::ldap_binddn,
+  $ldap_bindpw          = $::icinga::ldap_bindpw,
+  $ldap_userattr        = $::icinga::ldap_userattr,
+  $ldap_groupattr       = $::icinga::ldap_groupattr,
+  $ldap_filter_extra    = $::icinga::ldap_filter_extra,
+  $ldap_auth_group      = $::icinga::ldap_auth_group,
+  $cgi_url              = $::icinga::cgi_url,
+  $cgi_path             = $::icinga::cgi_path,
+  $html_path            = $::icinga::html_path,
+  $css_path             = $::icinga::css_path,
+  $pnp4nagios_html_path = $::icinga::pnp4nagios_html_path,
+) {
   class { '::apache':
     default_vhost => false,
     mpm_module    => false,
   }
 
-  class { '::apache::mod::prefork':
-  }
-
-  class { '::apache::mod::php':
-  }
-
-  class { '::apache::mod::rewrite':
-  }
-
-  class { '::apache::mod::cgi':
-  }
-
-  # check if we are running pgsql and fix port if it is set to default mysql port
-  if $icinga::params::web_db_server == 'pgsql' and $icinga::params::web_db_port == 3306 {
-    $web_db_port = 5432
-  } else {
-    $web_db_port = $icinga::params::web_db_port
-  }
-
-  # check if we are running pgsql and fix port if it is set to default mysql port
-  if $icinga::params::ido_db_server == 'pgsql' and $icinga::params::ido_db_port == 3306 {
-    $ido_db_port = 5432
-  } else {
-    $ido_db_port = $icinga::params::ido_db_port
-  }
```

Comment:
Yes i got the logic i just moved it to the erb file, personal preference just think it looks cleaner there and